Support child projects android

codingjeremy opened the pull request:
Add support for generating dependency reports to repos with child samples, i.e., repos where the gradle files aren't in the root directory.

samtstern commented on use-latest-deps-android.sh:
```diff
@@ -57,26 +107,32 @@ REPO=$1
-# Generate JSON dependencies report
-./gradlew dependencyUpdates -Drevision=release -DoutputFormatter=json
+generate_dependencies_report "${DIR}/"
+update_gradle_exit_code="$?"
```
nit: name this `generate_dependencies_report_exit_code` or something else that is more similar to the actual function name?

samtstern commented on use-latest-deps-android.sh:
```diff
@@ -57,26 +107,32 @@ REPO=$1
-# Remove the virtualenv
-rm -rf env
+  # Generate list of folders in current directory.
+  array=(*/)
```
nit: name this `child_dirs` not `array`?

codingjeremy commented:
I addressed all changes, but I want to roll this into my own master along with my python changes, so I'm closing the PR. I will submit new one with all changes (and updated to proper HEAD). Thanks for review!
